regions/core/attributes.py

```diff
@@ -165,3 +166,42 @@ def _validate(self, value):
         if not isinstance(value, self.regionclass):
             raise ValueError(f'{self.name!r} must be a '
                              f'{self.regionclass.__name__} object')
+
+
+class RegionMetaDescr(RegionAttribute):
+    """
+    Descriptor class for the region meta dictionary.
+
+    If input as a pure `dict`, it will be converted to a `RegionMeta`
+    object.
+    """
+    def __set__(self, instance, value):
+        # RegionMeta subclasses dict
+        if type(value) == dict:  # pylint: disable=C0123
```

Comment:
Is this intentionally not `isinstance(value, dict)`?  I think that can result in subclasses of dictionaries being allowed - which I think is something we want here.

Reply:
Yes, it's intentionally not `isinstance(value, dict)` because `RegionMeta` subclasses `dict`.  We only want to convert `dict` input into `RegionMeta`.  So it's either this or:
`if isinstance(value, dict) and not isinstance(value, RegionMeta):`.

Reply:
But I wasn't thinking about *other* subclasses of dict.  I've updated this.  Thanks!